### PR TITLE
Add `qmk license-check` developer-level CLI command.

### PR DESCRIPTION
**Pull request by tzarc**
## Description

As per the title. Initial RFC.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: lib/python/qmk/constants.py [hunk 1]
```diff
@@ -189,3 +189,126 @@
...
+        of the License, or (at your option) any later version.
+        """, """\
+        This program is free software; you can redistribute it and/or
+        modify it under the terms of the GNU General Public License
+        as published by the Free Software Foundation; either version 2
+        of the License, or any later version.
+        """
+        ]
+    ),
+    ('GPL-2.0-only', ["""\
```
Comment (drashna) [inline lib/python/qmk/constants.py:209]: This inconsistency in formatting bugs me.

**Review by drashna — APPROVED** [review]
`__attribute__((weak))` ✅

File: lib/python/qmk/constants.py (hunk 1 above)
Comment (tzarc) [reply lib/python/qmk/constants.py:209]: Same, but this is what the formatter spits out. Annoying.

Comment (sigprof) [reply lib/python/qmk/constants.py:209]: We could change the data structure, but even that does not make the formatting fully consistent between the cases with single and multiple values:
```python
LICENSE_TEXTS = {
    'GPL-2.0-or-later': [
        """\
        This program is free software; you can redistribute it and/or
        modify it under the terms of the GNU General Public License
        as published by the Free Software Foundation; either version 2
        of the License, or (at your option) any later version.
        """, """\
        This program is free software; you can redistribute it and/or
        modify it under the terms of the GNU General Public License
        as published by the Free Software Foundation; either version 2
        of the License, or any later version.
        """
    ],
    'GPL-2.0-only': ["""\
        This program is free software; you can redistribute it and/or
        modify it under the terms of the GNU General Public License as
        published by the Free Software Foundation; version 2.
        """],
```
If all else fails, you could slap `# yapf:disable` after the final closing bracket and format the value manually.

Comment (tzarc) [reply lib/python/qmk/constants.py:209]: I don't feel a strong enough need to fight the formatter. This is something that'll be touched only once every blue moon.

**Pull request merged**